docs/usage-guide/topics/ch11-resumption.md

```diff
@@ -25,6 +25,8 @@ Servers should set the three caching callback functions: `s2n_config_set_cache_s
 
 Clients should call `s2n_connection_get_session()` to retrieve some serialized state about the session. Then `s2n_connection_set_session()` should be called with that saved state when attempting to resume a new connection.
 
+Applications must perform a graceful TSL shutdown in order to be able to resume a session with session caching. If the connection results in abrupt termination (i.e, no `close_notify` alert is received by server), the session cache will be deleted and the session will not be available for resumption. Additionally, any errors during the connection will also result in the session cache being deleted. For more information on how to close connections, see [Closing the Connection](./ch07-io.md#closing-the-connection)
```

Comment:
This makes it sound like an abrupt termination is not an error. Would it maybe be more clear to phrase this like "any error encountered during the connection will result in the the session being evicted" and then "this includes an abrupt termination..."?